lib/elixir_analyzer/exercise_test/common_checks/debug_functions.ex

```diff
@@ -0,0 +1,32 @@
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.DebugFunctions do
+  @moduledoc """
+  This is an exercise analyzer extension module used for common tests lookin for debugging functions
+  """
+
+  alias ElixirAnalyzer.Comment
+  alias ElixirAnalyzer.Submission
+
+  # Suppressing common tests prevents this module to recursively call common tests
+  use ElixirAnalyzer.ExerciseTest, suppress_common_tests: true
```

Comment:
This doesn't look so good... circular dependencies between modules usually mean trouble. Is there a way to avoid this while still using `assert_no_call` for this common check?

I do have an idea and I'm proposing it without checking if it will work, so beware 😁 could we inject the calls to `assert_no_call` via `__using__` in `ElixirAnalyzer.ExerciseTest`? This module could define for example a macro like this instead of the `run` function that could be then used in `ElixirAnalyzer.ExerciseTest`:

```elixir
defmacro inject() do
  quote do
    assert_no_call "solution doesn't use IO.inspect" do
	  type :informational
	  comment ElixirAnalyzer.Constants.solution_debug_functions()
	  called_fn module: IO, name: :inspect
	end
  end
end
```

Reply:
Mmh, I didn't think about circular dependencies... True, it's not desirable.

I had another version working which didn't do that, but it involved basically reimplementing `ElixirAnalyzer.ExerciseTest ` in the file. It worked but wasn't pretty. 

I think I get what you're proposing. Something like this?

```elixir
defmodule ElixirAnalyzer.ExerciseTest do
  alias  [...]

  defmacro __using__(opts) do
    quote do
      use ElixirAnalyzer.ExerciseTest.CommonChecks # <== this injects extra features and calls 
      use ElixirAnalyzer.ExerciseTest.Feature
      use ElixirAnalyzer.ExerciseTest.AssertCall
      @suppress_tests unquote(opts)[:suppress_tests]

[...]
```
I'm too sleepy to investigate more right now, but I'll try soon :)

(PS: doesn't the GitHub highlighter know that `...` is a valid variable name? haha)

Reply:
Yeah, something like it. I guess we need some sort of split between common checks that work directly on the AST and common checks that inject `form` and `assert_call` macros.

Reply:
I came up with something relatively painlessly. It's using a lot of `__using__` macros, but it might be the price we have to pay? Unless you can think of something cleaner...

Reply:
It looks perfect to me now